pkg/resource/deploy/step_generator.go

```diff
@@ -1612,25 +1615,48 @@ func (sg *stepGenerator) generateStepsFromDiff(
 	prov plugin.Provider, goal *resource.Goal, randomSeed []byte,
 	autonaming *plugin.AutonamingOptions,
 ) ([]Step, bool, error) {
-	diff, pcs, err := sg.diff(
-		event, goal, autonaming, randomSeed,
-		urn, old, new, oldInputs, inputs, prov)
-	if pcs != nil {
-		return []Step{
-			NewDiffStep(sg.deployment, pcs, old, new, goal.IgnoreChanges),
-		}, true, nil
+	// We might be triggering a replace if we have an old state and it's trigger value is different, we don't
+	// even need to bother doing a diff in this case because we know this replacement. Note that we treat Null
+	// as a special case for "no trigger", that is if Trigger goes from Null to anything, or from anything to
+	// Null it does not cause a replace.
+	triggerReplace := !new.ReplacementTrigger.IsNull() && !old.ReplacementTrigger.IsNull() &&
+		!new.ReplacementTrigger.DeepEquals(old.ReplacementTrigger)
+	// TODO: THIS IS INTERESTING!!! We don't really want DeepEquals here because we could have unknowns so we
+	// actually have a tri-state here, of (equal, not-equal, unknown). If equal that's easy, we're not doing a
+	// replace, likewise if not-equal we _are_ doing a replace, but the unknown state is tricky because it
+	// means we _might_ be doing a replace! That results in a maybe_replace_maybe_update step which we don't
+	// currently have. This is _very_ similar to our loop support (index resource option) where we would end
+	// up with things like maybe_delete_maybe_update. For now we just handle this the same as
+	// replaceOnChanges, i.e. unknown means assume a replace.
+
+	var diff plugin.DiffResult
+	var pcs *promise.CompletionSource[plugin.DiffResult]
+	var err error
+	if triggerReplace {
+		// Return that there is a diff, but don't fill in any details.
```

Comment:
This comment could use a few words on *why* we don't fill in any details/why that is fine.